site/content/api/Command-And-Control-API.md

```diff
@@ -3,38 +3,138 @@ title = "Command & Control API"
+2. The *Business Application* has established an AMQP link in role *sender* with the target address `control/${tenant_id}/${device_id}`, where `${device_id}` is the ID of the device to send messages to and `${tenant_id}` is the ID of the tenant that the device belongs to.
+
+**Link establishment**
+
+The following sequence diagram shows the necessary links that need to be established:
+
+![Send Command Preconditions](../command_control_send_preconditions.png)
+
+## Preconditions for receiving a command response
+
+*Command* messages (in contrast to *notification command* messages) need to be responded by the device. For this purpose,
+the *Business Application* needs to open a receiver link as precondition: 
+
+1. The *Business Application* has established an AMQP link in role *receiver* with the source address `control/${tenant_id}/${reply-id}`.
+This link is used by the *Business Application* to receive the response of the executed command. This link’s source address is also referred to as `the reply-to address` for the request messages. 
+2. The `${reply-id}` may be any arbitrary string chosen by the client. 
+
+### Choosing a strategy for building the receiver link address
+
+While the `${reply-id}` can be completely chosen by the client, the specific use case for sending commands may influence 
+how it should be set.
+Thus following hints for choosing the right `${reply-id}` may be helpful:
+
+- if only one command response is expected from the device, the `${reply-id}` could be set to a UUID and closed again after receiving the response.
+- if several command responses are expected from the device (e.g. since multiple commands were or will be sent to it),  the `${reply-id}` could be set to the `${device_id}` plus an arbitrary string to scope the command sender instance. Such a link should usually remain open after the reception of any response. 
+- if all command responses for the `${tenant_id}` shall be received in the application,  the `${reply-id}` could be set to just an arbitrary (short) string to scope the command sender instance (without the `${device_id}`). Such a link should usually remain open after the reception of any response.
+
+
+**Link establishment**
+
+The following sequence diagram shows the necessary links that need to be established to enable sending back a response to the Business Application:
+
+![Send Command](../command_control_receive_preconditions.png)
+
+# Outcomes for sending a command or notification command
+Commands and notification commands that are forwarded by the AMQP 1.0 network to a protocol adapter need to be responded with one of the following outcomes:
+
+| Processing in protocol adapter   | AMQP outcome                  |
+|----------------------------------|------------------------------ |
+| Bad command                      | Rejected ("hono:bad-request") |
+| Device not connected (anymore)   | Release                       | 
+| Command delivered                | Accepted                      | 
```

Comment:
I think we should be a bit more explicit in what "command delivered" means. Is that a guarantee that the command did reach the device?

Reply:
Good point - it means "it reached the protocol adapter". More guarantees we cannot give, if an application needs to be absolutely sure that the command reached the device, it
- must not use a "notification command", but a "command"
- wait for the reply from the device

I will clarify this here that only the protocol adapter received the command.